lib/CompilerPluginSupport/CMakeLists.txt

```diff
@@ -36,11 +36,25 @@ if(SWIFT_SWIFT_PARSER)
       -enable-library-evolution;
       -emit-module-interface;
       -target;${target}>)
-
-  target_link_libraries("${library_name}"
-    PRIVATE
-    swiftCore
-  )
+      
+  # Ensure that we do not link the _StringProcessing module. But we can
+  # only pass this flag for new-enough compilers that support it.
+  file(WRITE "${CMAKE_BINARY_DIR}/tmp/empty-check-string-processing.swift" "")
+  execute_process(
+    COMMAND
+      "${CMAKE_Swift_COMPILER}"
+      -Xfrontend -disable-implicit-string-processing-module-import
+      -c - -o /dev/null
+    INPUT_FILE
+      "${CMAKE_BINARY_DIR}/tmp/empty-check-string-processing.swift"
+    OUTPUT_QUIET ERROR_QUIET
+    RESULT_VARIABLE
+      SWIFT_SUPPORTS_DISABLE_IMPLICIT_STRING_PROCESSING_MODULE_IMPORT)
+  if(NOT SWIFT_SUPPORTS_DISABLE_IMPLICIT_STRING_PROCESSING_MODULE_IMPORT)
+    add_compile_options(
```

Comment:
This is the same as #61889. Should I change this in ASTGen too?

Reply:
Not in this PR, it can be done at a later time -- I suspect in there it works because we call `add_library` after `add_compile_options`)

If you are concerned about having different code paths, you can move this section before `add_library`, and tackle changing the CMake command used at a later time

Reply:
https://github.com/apple/swift/pull/62155